common/subdevice-model.cpp

```diff
@@ -612,63 +626,287 @@ namespace rs2
+                                if (res >= RS2_SENSOR_MODE_VGA && res < RS2_SENSOR_MODE_COUNT)
+                                {
+                                    try
+                                    {
+                                        s->set_option(RS2_OPTION_SENSOR_MODE, float(res));
+                                    }
+                                    catch (const error& e)
+                                    {
+                                        error_message = error_to_string(e);
+                                    }
+
+                                    // Only update the cached value once set_option is done! That way, if it doesn't change anything...
+                                    try
+                                    {
+                                        int sensor_mode_val = static_cast<int>(s->get_option(RS2_OPTION_SENSOR_MODE));
+                                        {
+                                            ui.selected_res_id = get_resolution_id_from_sensor_mode(
+                                                static_cast<rs2_sensor_mode>(sensor_mode_val),
+                                                res_values);
+                                        }
+                                    }
+                                    catch (...) {}
+                                }
+                                else
+                                {
+                                    error_message = rsutils::string::from() << "Resolution " << width << "x" << height
+                                        << " is not supported on this device";
+                                }
+                            }
+                            else
+                            {
+                                ui.selected_res_id_map[stream_type_id] = tmp_selected_res_id;
+                            }
+                        }
+                        ImGui::PopStyleColor();
+                        ImGui::PopItemWidth();
+                    }
+                }
+                ImGui::SetCursorPosX(col0);
             }
 
-            // Draw combo-box with all format options for current device
+            // stream and formats
+            // Draw combo-box with all format options for current stream type
+            std::map<rs2_stream, std::vector<int>> stream_to_index;
+            int depth_res_id, ir1_res_id, ir2_res_id;
+            get_depth_ir_mismatch_resolutions_ids(depth_res_id, ir1_res_id, ir2_res_id);
+            stream_to_index[RS2_STREAM_DEPTH] = { depth_res_id };
+            stream_to_index[RS2_STREAM_INFRARED] = { ir1_res_id, ir2_res_id };
+
             for (auto&& f : formats)
             {
                 // Format
                 if (f.second.size() == 0)
                     continue;
 
+                if (std::find(stream_to_index[stream_type].begin(), stream_to_index[stream_type].end(), f.first) == stream_to_index[stream_type].end())
```

Comment:
Just curious..
Why not `stream_to_index[stream_type].find(f.first) != stream_to_index[stream_type].end()`
Probably the same..

Reply:
opposite, but yes, seems more readable.

Reply:
Correct, my bad